From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ROCKNIX <rocknix@rocknix.org>
Date: Tue, 04 Mar 2026 00:00:00 +0000
Subject: [PATCH] arm64: dts: rockchip: rk3566: enable DMC devfreq on Powkiddy
 X55

Enable the DMC devfreq driver on the Powkiddy X55 (and X35S which
includes this DTS) by providing the center-supply regulator binding
and setting status to "okay".

Signed-off-by: ROCKNIX <rocknix@rocknix.org>
---
 .../boot/dts/rockchip/rk3566-powkiddy-x55.dts    | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3566-powkiddy-x55.dts b/arch/arm64/boot/dts/rockchip/rk3566-powkiddy-x55.dts
index XXXXXXX..XXXXXXX 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-powkiddy-x55.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3566-powkiddy-x55.dts
@@ -980,3 +980,8 @@
 		remote-endpoint = <&dsi0_in_vp1>;
 	};
 };
+
+&dmc {
+	center-supply = <&vdd_logic>;
+	status = "okay";
+};
-- 
2.34.1

